oa4mp/serve.go

```diff
@@ -306,6 +306,28 @@ func ConfigureOA4MP() (launcher daemon.Launcher, err error) {
 		return
 	}
 
+	// Ensure the OA4MP storage directory exists and has correct permissions and ownership
+	OA4MPStoragePath := "/opt/scitokens-server/var/storage"
+	if _, err := os.Stat(OA4MPStoragePath); os.IsNotExist(err) {
+		log.Debugln("OA4MP storage directory does not exist. Creating: ", OA4MPStoragePath)
+		if err = os.MkdirAll(OA4MPStoragePath, 0755); err != nil {
+			log.Errorln("Failed to create directory:", OA4MPStoragePath, ": ", err)
+		}
+	}
+
+	if err := os.Chmod(OA4MPStoragePath, 0700); err != nil {
+		log.Errorln("Failed to change permissions of: ", OA4MPStoragePath, ": ", err)
```

Comment:
```suggestion
		log.Errorln("Failed to change permissions of", OA4MPStoragePath, ":", err)
```

also return on error.